### PR TITLE
[SFP] Change call of sx_mgmt_phy_module_info_get() SDK API to sysfs

### PR DESCRIPTION
**Pull request by vadymhlushko-mlnx**
#### Why I did it
SDK/FW team implemented a Linux SYSFS for the module operational state and status error fields.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Modify the `platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py` file by change the call of sx_mgmt_phy_module_info_get() SDK API to sysfs

#### How to verify it
1. Simulate the `unplug` cable event
2. Check the CLI output
    - `sfputil show presence`
    - `sfputil show error-status -hw`
3. Simulate the `plug` cable event
4. Repeat 2 step

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 1]
```diff
@@ -230,21 +244,18 @@ def _get_module_info(self, sdk_handle, sdk_index):
...
-        sx_mgmt_module_id_info_t_arr_setitem(module_id_info_list, 0, module_id_info)
-
-        rc = sx_mgmt_phy_module_info_get(sdk_handle, module_id_info_list, 1, module_info_list)
-        assert SX_STATUS_SUCCESS == rc, "sx_mgmt_phy_module_info_get failed, error code {}".format(rc)
-
-        mod_info = sx_mgmt_phy_module_info_t_arr_getitem(module_info_list, 0)
-        return mod_info.module_state.oper_state, mod_info.module_state.error_type
+            The error code fetch from sysfs
+        """
+        sysfs_dir = "/sys/module/sx_netdev"
```
Comment (vadymhlushko-mlnx) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:249]: move to the beginning of the file

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 2]
```diff
@@ -230,21 +244,18 @@ def _get_module_info(self, sdk_handle, sdk_index):
...
+        logical_ports = self.platform_sfputil.physical_to_logical[sdk_index + 1]
+        status_file = os.path.join(sysfs_dir, logical_ports[0], 'module', 'status')
```
Comment (vadymhlushko-mlnx) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:251]: consider to use constants as for `sysfs_dir`

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 3]
```diff
@@ -211,6 +217,17 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 
 class NvidiaSFPCommon(SfpOptoeBase):
+    platform_sfputil = sonic_platform_base.sonic_sfp.sfputilhelper.SfpUtilHelper()
```
Comment (Junchao-Mellanox) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: Hi Vadym, we requested sdk to provide module based sysfs, we don't use logical port based sysfs here. Please use sysfs /sys/module/sx_core/asic0/moduleX/status and /sys/module/sx_core/asic0/moduleX/statuserror here. In that case, you don't need this platform_sfputil anymore.

**Review by Junchao-Mellanox — CHANGES_REQUESTED** [review]

**Comment by Junchao-Mellanox** [comment]
Could you please add unit test for this change?

**Comment by vadymhlushko-mlnx** [comment]
> Could you please add unit test for this change?

UT already implemented:
[test_sfp_get_error_status](https://github.com/sonic-net/sonic-buildimage/blob/bd413d20d2e264e2580e73867cd4cf3f097868b9/platform/mellanox/mlnx-platform-api/tests/test_sfp.py#L61)
[test_sfp_get_presence](https://github.com/sonic-net/sonic-buildimage/blob/bd413d20d2e264e2580e73867cd4cf3f097868b9/platform/mellanox/mlnx-platform-api/tests/test_sfp.py#L211)

**Comment by Junchao-Mellanox** [comment]
> > Could you please add unit test for this change?
> 
> UT already implemented: [test_sfp_get_error_status](https://github.com/sonic-net/sonic-buildimage/blob/bd413d20d2e264e2580e73867cd4cf3f097868b9/platform/mellanox/mlnx-platform-api/tests/test_sfp.py#L61) [test_sfp_get_presence](https://github.com/sonic-net/sonic-buildimage/blob/bd413d20d2e264e2580e73867cd4cf3f097868b9/platform/mellanox/mlnx-platform-api/tests/test_sfp.py#L211)

Those two cases cannot test the new code you added/updated.

**Review by Junchao-Mellanox — APPROVED** [review]

**Pull request closed without merging**